Std/Tactic/Basic.lean

```diff
@@ -1,165 +1,15 @@
-```lean
-iterate
-  tac₁
-  tac₂
-  ⋯
-```
--/
-syntax "iterate" (ppSpace num)? ppSpace tacticSeq : tactic
-macro_rules
-  | `(tactic| iterate $seq:tacticSeq) =>
-    `(tactic| try ($seq:tacticSeq); iterate $seq:tacticSeq)
-  | `(tactic| iterate $n $seq:tacticSeq) =>
-    match n.1.toNat with
-    | 0 => `(tactic| skip)
-    | n+1 => `(tactic| ($seq:tacticSeq); iterate $(quote n) $seq:tacticSeq)
-
-/--
-`repeat' tac` runs `tac` on all of the goals to produce a new list of goals,
-then runs `tac` again on all of those goals, and repeats until `tac` fails on all remaining goals.
--/
-elab "repeat' " tac:tacticSeq : tactic => do
-  setGoals (← repeat' (evalTacticAtRaw tac) (← getGoals))
-
-/--
-`repeat1' tac` applies `tac` to main goal at least once. If the application succeeds,
-the tactic is applied recursively to the generated subgoals until it eventually fails.
--/
-elab "repeat1' " tac:tacticSeq : tactic => do
-  setGoals (← repeat1' (evalTacticAtRaw tac) (← getGoals))
-
-/-- `subst_eqs` applies `subst` to all equalities in the context as long as it makes progress. -/
-elab "subst_eqs" : tactic => Elab.Tactic.liftMetaTactic1 (·.substEqs)
-
-/-- `split_ands` applies `And.intro` until it does not make progress. -/
-syntax "split_ands" : tactic
-macro_rules | `(tactic| split_ands) => `(tactic| repeat' refine And.intro ?_ ?_)
-
-/--
-`fapply e` is like `apply e` but it adds goals in the order they appear,
-rather than putting the dependent goals first.
--/
-elab "fapply " e:term : tactic =>
-  evalApplyLikeTactic (·.apply (cfg := {newGoals := .all})) e
-
-/--
-`eapply e` is like `apply e` but it does not add subgoals for variables that appear
-in the types of other goals. Note that this can lead to a failure where there are
-no goals remaining but there are still metavariables in the term:
-```
-example (h : ∀ x : Nat, x = x → True) : True := by
-  eapply h
-  rfl
-  -- no goals
--- (kernel) declaration has metavariables '_example'
-```
--/
-elab "eapply " e:term : tactic =>
-  evalApplyLikeTactic (·.apply (cfg := {newGoals := .nonDependentOnly})) e
-
-/--
-Tries to solve the goal using a canonical proof of `True`, or the `rfl` tactic.
-Unlike `trivial` or `trivial'`, does not use the `contradiction` tactic.
--/
-macro (name := triv) "triv" : tactic =>
-  `(tactic| first | exact trivial | rfl | fail "triv tactic failed")
-
-/-- `conv` tactic to close a goal using an equality theorem. -/
-macro (name := Conv.exact) "exact " t:term : conv => `(conv| tactic => exact $t)
+/-!
+# Common tactics used throughout Std.
 
-/-- The `conv` tactic `equals` claims that the currently focused subexpression is equal
- to the given expression, and proves this claim using the given tactic.
-```
-example (P : (Nat → Nat) → Prop) : P (fun n => n - n) := by
-  conv in (_ - _) => equals 0 =>
-    -- current goal: ⊢ n - n = 0
-    apply Nat.sub_self
-  -- current goal: P (fun n => 0)
-```
+This is an import only file.
```

Comment:
Some of the style linters use a regex-based identification of import-only files, so I would recommend not putting any module docs in them. Use regular comments if needed.